### PR TITLE
Fix AWS security groups

### PR DESCRIPTION
**Pull request by flavio**
## Why is this PR needed?

The deployment of CaaSP on top of AWS is not working due to the security groups rules being wrong.

While this is not yet a supported deployment platform, I'm working on the delivery of a workshop demoing CaaSP on AWS.

## What does this PR do?

This PR does the following actions:

  * Remove rules that are no longer needed by v4
  * Add missing rules

## Anything else a reviewer needs to know?

This is a first iteration, once this code is merged I'll file another PR that organizes the rules in the same way  as other deployments do. I didn't do that right now to make life easier to reviewers.

Reviewing the whole diff at the same time can be a bit confusing. I highly recommend to go over each individual commit: they are well scoped and have nice explanation messages. They will make your life easier.

A similar work has to be done for OpenStack, I'll resume the work done by @ereslibre [here](https://github.com/SUSE/skuba/pull/414).

## Info for QA

This deployment is not yet supported by us, hence there's no QA to be done.

However, if someone is really interested, the steps to verify the fix are really easy (see below).

### Steps to reproduce the bug

  * Deploy the AWS infra: one master and one worker nodes are enough
  * Deploy CaaSP beta 5 on top of it
  * Check the output of `kubectl get pods --namespace kube-system`: all the pods should be running fine (without the patch some of them enter a backoff state because they never ever start properly)

### Related info

While working on this PR I realized our documentation was outdated. That resulted in the creation of [this](https://github.com/SUSE/doc-caasp/pull/398) PR against the CaaSP documentation. Feel free to take a look at it to have a better understanding of the ports that are required.

### Status **BEFORE** applying the patch

PODs like the cilium, cilium-operator stayed in persistent failure mode.

### Status **AFTER** applying the patch

All the PODs are running fine.

### CODE REVIEW

File: ci/infra/aws/security-groups.tf [hunk 1]
```diff
@@ -102,96 +113,120 @@ resource "aws_security_group" "allow_control_plane_traffic" {
...
     to_port     = 2380
     protocol    = "tcp"
     cidr_blocks = ["${var.vpc_cidr_block}"]
   }
 
+  # VXLAN traffic (cilium) - internal
   ingress {
-    from_port   = 8285
-    to_port     = 8285
+    from_port   = 8471
```
Comment (ereslibre) [inline ci/infra/aws/security-groups.tf]: According to https://docs.cilium.io/en/v1.4/install/system_requirements/#firewall-rules, `8472` should suffice. Am I missing something? [1]

File: ci/infra/aws/security-groups.tf [hunk 2]
```diff
@@ -102,96 +113,120 @@ resource "aws_security_group" "allow_control_plane_traffic" {
...
 
+  tags = "${merge(local.tags, map(
+    "Name", "${var.stack_name}-control-plane",
+    "Class", "SecurityGroup"))}"
+
+  # VXLAN traffic (cilium) - internal
   ingress {
-    from_port   = 2380
-    to_port     = 2380
+    from_port   = 8471
```
Comment (ereslibre) [inline ci/infra/aws/security-groups.tf]: Same as [1]

File: ci/infra/aws/security-groups.tf [hunk 3]
```diff
@@ -102,96 +113,120 @@ resource "aws_security_group" "allow_control_plane_traffic" {
...
+    to_port     = 8472
+    protocol    = "udp"
+    cidr_blocks = ["${var.vpc_cidr_block}"]
+  }
+
+  # CRI-O streaming - internal only
```
Comment (ereslibre) [inline ci/infra/aws/security-groups.tf]: Do we really need the CRI-O streaming feature exposed?

File: ci/infra/aws/security-groups.tf [hunk 4]
```diff
@@ -208,6 +243,14 @@ resource "aws_security_group" "elb" {
     "Class", "SecurityGroup"))}"
 
   # HTTP access from anywhere
+  ingress {
+    from_port   = 80
+    to_port     = 80
```
Comment (ereslibre) [inline ci/infra/aws/security-groups.tf]: Is this in preparation for the future ingress controller support? [2]

File: ci/infra/aws/security-groups.tf [hunk 5]
```diff
@@ -208,6 +243,14 @@ resource "aws_security_group" "elb" {
...
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  # HTTPS access from anywhere
```
Comment (ereslibre) [inline ci/infra/aws/security-groups.tf]: Same question as [2]

File: ci/infra/aws/security-groups.tf (hunk 4 above)
Comment (flavio) [reply ci/infra/aws/security-groups.tf]: yes

File: ci/infra/aws/security-groups.tf (hunk 5 above)
Comment (flavio) [reply ci/infra/aws/security-groups.tf]: yes

File: ci/infra/aws/security-groups.tf (hunk 3 above)
Comment (flavio) [reply ci/infra/aws/security-groups.tf]: I don't think so, I'm discussing that over there: https://github.com/SUSE/doc-caasp/pull/398

File: ci/infra/aws/security-groups.tf (hunk 1 above)
Comment (flavio) [reply ci/infra/aws/security-groups.tf]: Yeah, this is odd. @mrostecki should we drop port `8471`?

Comment (flavio) [reply ci/infra/aws/security-groups.tf]: Dropped

File: ci/infra/aws/security-groups.tf (hunk 3 above)
Comment (flavio) [reply ci/infra/aws/security-groups.tf]: Dropped

**Comment by flavio** [comment]
I've updated the PR to reflect what emerged inside of https://github.com/SUSE/doc-caasp/pull/398

**Comment by mssola** [comment]
What's the status of this?

**Comment by flavio** [comment]
> What's the status of this?

We successfully update the list of network ports that are required by CaaSP v4. The PR has been updated to reflect that.

It would be great if we could merge it. I've more fixes coming up for AWS.

I admit none of that is relevant for v4 GM, but:

  * They are needed because of an upcoming event
  * All the changes are scoped to the AWS deployment, which is not part of the skuba package (the whole directory is just ignored).

@jordimassaguerpla: would you be against merging the PR now?

**Comment by jordimassaguerpla** [comment]
I am fine. We need 2 more and fix the tests please. Thanks.

**Pull request merged**